src/Features/Core/Portable/Diagnostics/Analyzers/IDEDiagnosticIds.cs

```diff
@@ -86,6 +86,8 @@ internal static class IDEDiagnosticIds
 
         public const string UseCompoundAssignmentDiagnosticId = "IDE0054";
 
+        public const string AsyncMainNotAvailableDiagnosticId = "IDE0055";
```

Comment:
Should this be below "Analyzer error Ids" and have an ID of `IDE1009` to differentiate it from usual IDE features? I know that this is not actually an analyzer error, but the group below also contains other things that are not analyzer errors, such as "UnboundIdentifierId", which is really similar to this feature in that it tries to substitute the work of the compiler.